### PR TITLE
OAuth: Fix oauth token sync being called on every request - generic oauth

### PR DESCRIPTION
**Pull request by Jguer**
OAuth token sync is being called on every request for generic_oauth and when the user is not using oauth because of `currentOAuthInfo.UseRefreshToken` being disabled or the auth info not being found. 

Give a 10 min break to the hook to avoid calling the auth info store 8 times every time a page is changed

### CODE REVIEW

**Review by kalleep — APPROVED** [review]

**Review by mgyongyosi — APPROVED** [review]

File: pkg/services/authn/authnimpl/sync/oauth_token_sync.go
Comment (gamab) [inline pkg/services/authn/authnimpl/sync/oauth_token_sync.go:91]: I'm a bit puzzled about this one. What would be this case? THe user has an `OAuthEntry` but it's comes from no known provider?

**Review by gamab — APPROVED** [review]
@linoman fyi for #79569

File: pkg/services/authn/authnimpl/sync/oauth_token_sync.go
Comment (kalleep) [reply pkg/services/authn/authnimpl/sync/oauth_token_sync.go:91]: This could happen if user has a entry to a oauth provider that used to be configured but no longer are

Comment (gamab) [reply pkg/services/authn/authnimpl/sync/oauth_token_sync.go:91]: Wow, thanks, talking about an edge case :smile:

**Comment by linoman** [comment]
@Jguer will it be possible to hold this PR until we merge this [PR](https://github.com/grafana/grafana/pull/79569)?

The PR I'm working on is almost finished but this PR will certainly create some conflicts.

**Comment by Jguer** [comment]
Can hold on it for sure

**Comment by Jguer** [comment]
No longer necessary

**Pull request closed without merging**
